pkg/scheduler/frameworkext/helper/synced_eventhandler.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"reflect"
+
+	"k8s.io/client-go/tools/cache"
+)
+
+type forceSyncEventHandler struct {
+	handler cache.ResourceEventHandler
+	syncCh  chan struct{}
+}
+
+func newForceSyncEventHandler(handler cache.ResourceEventHandler) *forceSyncEventHandler {
+	return &forceSyncEventHandler{
+		handler: handler,
+		syncCh:  make(chan struct{}, 1),
+	}
+}
+
+func (h *forceSyncEventHandler) syncDone() {
+	close(h.syncCh)
+}
+
+func (h *forceSyncEventHandler) waitForSyncDone() {
+	<-h.syncCh
+}
+
+func (h *forceSyncEventHandler) OnAdd(obj interface{}) {
+	h.waitForSyncDone()
+	h.handler.OnAdd(obj)
+}
+
+func (h *forceSyncEventHandler) OnUpdate(oldObj, newObj interface{}) {
+	h.waitForSyncDone()
+	h.handler.OnUpdate(oldObj, newObj)
+}
+
+func (h *forceSyncEventHandler) OnDelete(obj interface{}) {
+	h.waitForSyncDone()
+	h.handler.OnDelete(obj)
+}
+
+func (h *forceSyncEventHandler) addDirectly(obj interface{}) {
+	h.handler.OnAdd(obj)
+}
+
+type CacheSyncer interface {
+	Start(stopCh <-chan struct{})
+	WaitForCacheSync(stopCh <-chan struct{}) map[reflect.Type]bool
+}
+
+// ForceSyncFromInformer ensures that the EventHandler will synchronize data immediately after registration,
+// helping those plugins that need to build memory status through EventHandler to correctly synchronize data
+func ForceSyncFromInformer(stopCh <-chan struct{}, cacheSyncer CacheSyncer, informer cache.SharedInformer, handler cache.ResourceEventHandler) {
```

Comment:
we may not need cacheSyncer arguments, we can use cache.WaitForCacheSync(stopCh, informer.HasSynced)

Reply:
if this function is waitForEventHandlerSync, we man can use it as follows? this may be better?

syncEventHandler := newForceSyncEventHandler(handler)
syncEventHandler.AddToInformer(informer)
syncEventHandler.waitForEventHandlerSync

Reply:
We need cacheSyncer  because we should use cacheSyncer start infromer to keep the syncer internal state. e.g.  informers.SharedInformerFactory.Start will keep the started state of informer internal.